src/com/facebook/buck/android/AndroidInstrumentationTestDescription.java

```diff
@@ -0,0 +1,125 @@
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.facebook.buck.android;
+
+import com.facebook.buck.model.BuildTarget;
+import com.facebook.buck.model.HasSourceUnderTest;
+import com.facebook.buck.rules.BuildRule;
+import com.facebook.buck.rules.BuildRules;
+import com.facebook.buck.rules.BuildRuleParams;
+import com.facebook.buck.rules.BuildRuleResolver;
+import com.facebook.buck.rules.BuildRuleType;
+import com.facebook.buck.rules.Description;
+import com.facebook.buck.rules.InstallableApk;
+import com.facebook.buck.rules.Label;
+import com.facebook.buck.rules.SourcePathResolver;
+import com.facebook.buck.util.HumanReadableException;
+import com.facebook.infer.annotation.SuppressFieldNotInitialized;
+import com.google.common.base.Optional;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.ImmutableSortedSet;
+import com.google.common.collect.Iterables;
+
+public class AndroidInstrumentationTestDescription
+    implements Description<AndroidInstrumentationTestDescription.Arg> {
+
+  public static final BuildRuleType TYPE = BuildRuleType.of("android_instrumentation_test");
+
+  private final Optional<Long> testRuleTimeoutMs;
+
+  public AndroidInstrumentationTestDescription(Optional<Long> testRuleTimeoutMs) {
+    this.testRuleTimeoutMs = testRuleTimeoutMs;
+  }
+
+  @Override
+  public BuildRuleType getBuildRuleType() {
+    return TYPE;
+  }
+
+  @Override
+  public Arg createUnpopulatedConstructorArg() {
+    return new Arg();
+  }
+
+  @Override
+  public <A extends Arg> AndroidInstrumentationTest createBuildRule(
+      BuildRuleParams params,
+      BuildRuleResolver resolver,
+      A args) {
+    BuildRule baseApk = resolver.getRule(args.apk);
+    if (!(baseApk instanceof InstallableApk)) {
+      throw new HumanReadableException(
+          "In %s, apk='%s' must be an android_binary() or apk_genrule() but was %s().",
+          params.getBuildTarget(),
+          baseApk.getFullyQualifiedName(),
+          baseApk.getType());
+    }
+
+    BuildRule instrumentationApk = resolver.getRule(args.instrumentationApk);
+    if (!(instrumentationApk instanceof AndroidInstrumentationApk)) {
+      throw new HumanReadableException(
+          "In %s, apk='%s' must be an android_instrumentation_apk() but was %s().",
+          params.getBuildTarget(),
+          instrumentationApk.getFullyQualifiedName(),
+          instrumentationApk.getType());
+    }
+
+    return new AndroidInstrumentationTest(
+        params.appendExtraDeps(
+                BuildRules.getExportedRules(
+                    Iterables.concat(
+                        params.getDeclaredDeps(),
+                        resolver.getAllRules(args.deps.get())))),
+        new SourcePathResolver(resolver),
+        (InstallableApk) baseApk,
+        (AndroidInstrumentationApk) instrumentationApk,
+        args.testRunner.or("android.test.InstrumentationTestRunner"),
```

Comment:
@tdrhq Thanks for the suggestion, that makes sense.  I modeled it after tryToExtractPackageName that is already in the AdbHelper.